src/Persistence/SlotEntityContentRepository.php

```diff
@@ -24,12 +23,19 @@ public function __construct(
 		private Authority $authority,
 		private WikiPageFactory $pageFactory,
 		private EntityTitleLookup $entityTitleLookup,
-		private string $slotName
+		private string $slotName,
+		private RevisionLookup $revisionLookup
 	) {
 	}
 
-	public function getContent( EntityId $entityId ): ?Content {
-		$revision = $this->getWikiPage( $entityId )?->getRevisionRecord();
+	public function getContent( EntityId $entityId, int $revisionId = 0 ): ?Content {
+		$page = $this->getWikiPage( $entityId );
+
+		if ( $page === null ) {
+			return null;
+		}
+
+		$revision = $this->revisionLookup->getRevisionByTitle( $page, $revisionId );
```

Comment:
This is where the `0`/`null` choice mentioned above comes in.

Not sure about performance difference of the lookup versus using the previous `getRevisionRecord()`. We could instead have an if here to use that if no `$revisionId` is specified. The current code is because it's less code.